extension.js

```diff
@@ -163,12 +179,87 @@ class MprisLabel extends PanelMenu.Button {
 				this.player = this.players.next();
 				this._refresh();
 				break;
+			case 'volume-up':
+				this._changeVolume(1);
+				break;
+			case 'volume-down':
+				this._changeVolume(-1);
+				break;
+		}
+	}
+
+	_changeVolume(delta){
+		let stream = [];
+		stream[0] = this.volumeControl.get_default_sink();
+		let stream_name = 'System Volume (Global)';
+
+		const CONTROL_SCHEME = this.settings.get_string('volume-control-scheme');
+
+		if(CONTROL_SCHEME == 'application' && this.player){
+			stream = this.stream;
+			if (!stream)
+				stream = this._getStream()
```

Comment:
`stream` is initialised with the fallback value higher up. Does this do anything?

Reply:
Ok, I got it now as it is being redefined in the row above... This will need to be replaced by `if(!stream.length)` now

Reply:
All or most of the checks against empty arrays that we do are done with `.length == 0` and `.length === 0`

Reply:
ok, I'll update accordingly